src/book/02-system/01-nubots/03-nuclear.mdx

```diff
@@ -5,3 +5,144 @@ title: NUClear
+
+```cpp
+on<<Trigger<A>, Optional<Trigger<B>>().then([](const A& a, const B& b) {
+
+});
+```
+
+In the above request, the trigger on data type B has been listed as optional, while the trigger for data type A is listed as mandatory. Yet the callback function lists arguments for both A and B.
+
+Let's say that type A is emitted to the PowerPlant, but at this time, type B does not have any data associated with it.
+
+Even though type B is missing data, since it's listed as optional, the task associated with this reaction can be scheduled. When executing the callback for the reaction, NUClear will identify that type B is not present, and will remove reference to this type from the callback, so that the task is only run with type A.
+
+Effectively, through the application of fission, the callback is restructured to look like the following example.
+
+```cpp
+.then([](const A& a){
+
+});
+```
+
+## Emit Statements
+
+Emit statements are used by Reactors wanting to emit data to the PowerPlant. When using NUClear, data will most likely be emitted during a reaction. However, where necessary, emissions can also occur during reactor construction, or in some cases from within the PowerPlant itself.
+
+Any data emitted to the PowerPlant will be sent with a unique pointer. The PowerPlant will take ownership of this pointer and run any necessary callbacks to trigger reactions.
+
+Data can be emitted under varying scopes. These can be local or network emissions. More can be found on emit statements in the [NUClear documentation](https://nuclear.readthedocs.io/en/latest/dsl.html#emit-statements).
+
+### Local Emissions
+
+Local emissions send data to the local instance of the NUClear PowerPlant. Essentially this is the current running process. There are a number of scopes under which these emissions can take place. The syntax for these emissions are `emit<Scope::SCOPE>(data, dataType);`, where `SCOPE` is replaced with a scope as defined below.
+
+| Scope      | Description                                                                                                                                  |
+| ---------- | -------------------------------------------------------------------------------------------------------------------------------------------- |
+| LOCAL      | When emitting data under this scope, tasks are distributed via the thread pool for execution. This is the default behaviour when emitting.   |
+| DIRECT     | When emitting data under this scope, the tasks created as a result of this emission will bypass the threadpool, and be executed immediately. |
+| INITIALISE | This scope emits data as the system starts up.                                                                                               |
+| DELAY      | This will emit data, after the provided delay. The syntax includes a delay time `emit<Scope::DELAY>(data, delay(ticks), dataType);`.         |
```

Comment:
There is no dataType on the end
```suggestion
| DELAY      | This will emit data, after the provided delay. The syntax includes a delay time `emit<Scope::DELAY>(data, delay(ticks));`.         |
```